pkg/server/singleprocess/service_ui_pipeline.go

```diff
@@ -0,0 +1,81 @@
+package singleprocess
+
+import (
+	"context"
+
+	"github.com/hashicorp/go-hclog"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	pb "github.com/hashicorp/waypoint/pkg/server/gen"
+	"github.com/hashicorp/waypoint/pkg/server/hcerr"
+	serverptypes "github.com/hashicorp/waypoint/pkg/server/ptypes"
+)
+
+func (s *Service) UI_ListPipelines(
```

Comment:
Might be nice to add a note somewhere saying that, while paginating is part of the request, this doesn't yet support pagination and will return everything every time.